docs/resource-validation.md

```diff
@@ -15,10 +15,17 @@ created.
 
 A Gateway API resource (a new resource or an update for the existing one) is validated by the following steps:
 
+### For Kubernetes 1.25+
```

Comment:
same comment as @kate-osborn  in the other doc for webhook validation done by NGF in this doc -- "I think keeping this section in the documentation makes sense since the code is still performing these additional checks, and we don't (yet) have plans to remove them."